Replace dom-scroll-into-view with native methods.

sgomes opened the pull request:
The usage that is made of `dom-scroll-into-view` (a 3rd party `npm` module) in Calypso is all achievable through native methods on the web platform. Making the replacement saves us about 1.6KB (compressed) in the critical path.

This feature should work in older browsers, with some amount of graceful degradation. Namely, elements may always scroll to the top, even if scrolling is not necessary. I believe this is acceptable, as the core functionality remains intact.

#### Changes proposed in this Pull Request

* Rewrite usage of `dom-scroll-into-view` with native `scrollIntoViewIfNeeded ` and `scrollIntoView`.

#### Testing instructions

To test site selector:

* Click "Switch sites" to open the site selector
* Shrink your browser window vertically so that not all elements fit and scrolling is needed
* Scroll through elements with up and down arrow keys on your keyboard, and ensure that the list scrolling behaviour is correct, with items popping into view as you scroll.


To test token field:

* Go to "Manage" -> "Domains"
* Click "Add Domain"
* Start typing in the text field
* Once a list of results comes up, click "More extensions"
* Click the text field (it has the placeholder text "Select an extension") to focus it
* Scroll through elements with up and down arrow keys on your keyboard, and ensure that the list scrolling behaviour is correct, with items popping into view as you scroll.

sgomes committed:
Replace dom-scroll-into-view with native methods.

The usage of `dom-scroll-into-view` in Calypso is all achievable through
native methods on the web platform.

sgomes commented:
Looks like bundle sizes didn't go down, because `dom-scroll-into-view` is still being used in `@wordpress/components` :-/

Still, the benefits of dropping an `npm` package for a native feature are probably enough, and I'll see what I can do about removing `dom-scroll-into-view` from `@wordpress/components` separately.

lynnbgriffith reviewed (commented):
I noticed that we already have a `scrollIntoViewport` function (https://github.com/Automattic/wp-calypso/blob/f6371e434a435301eeaf09eeba4b2d1e77f2cecc/client/lib/scroll-into-viewport/index.js).  It looks like your approach here -- leveraging the native `scrollIntoViewIfNeeded`) could be a slight improvement over that.  It might also be nice to have the feature detection conditionals you have in this PR encapsulated into a reusable helper function anyway.

sgomes commented:
> I noticed that we already have a `scrollIntoViewport` function (https://github.com/Automattic/wp-calypso/blob/f6371e434a435301eeaf09eeba4b2d1e77f2cecc/client/lib/scroll-into-viewport/index.js). It looks like your approach here -- leveraging the native `scrollIntoViewIfNeeded`) could be a slight improvement over that. It might also be nice to have the feature detection conditionals you have in this PR encapsulated into a reusable helper function anyway.

Great catch, @griffbrad ! I'll see what I can do about unifying this under that lib 👍

sgomes committed:
Rewrite lib/scroll-into-viewport.

It now matches the Element.scrollIntoView() API, including the proposed
`scrollMode` option.

sgomes commented:
> Great catch, @griffbrad ! I'll see what I can do about unifying this under that lib 👍

@griffbrad I've now rewritten `lib/scroll-into-viewport` to match the `Element.scrollIntoView` API, and updated all of its consumers to the new API. 👍

lynnbgriffith reviewed (approved):
🚢 LGTM

sgomes committed:
Fix refs in form input.

sgomes commented:
> 🚢 LGTM

Woohoo, thanks for the review! I'll leave this for Monday in order to avoid potential breakage over the weekend 😄

sgomes committed:
Fix refs in StateSelect.
